### PR TITLE
Introduce Wiki markdown API option

### PR DESCRIPTION
**Pull request by andrew-boyarshin**
As it is in the title.

For go-gitea/gitea#1027.

### CODE REVIEW

**Comment by lunny** [comment]
LGTM

**Comment by andrew-boyarshin** [comment]
This PR would also require documentation change. For example:

Wiki: ```"true"```/```"false"```, optional, false by default

Render markdown as wiki page. Influences link URLs.

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

File: gitea/miscellaneous.go [hunk 1]
```diff
@@ -9,4 +9,5 @@ type MarkdownOption struct {
 	Text    string
 	Mode    string
 	Context string
+	Wiki    string
```
Comment (bkcsoft) [inline gitea/miscellaneous.go]: This should be a `bool`, not a string

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: gitea/miscellaneous.go (hunk 1 above)
Comment (lunny) [reply gitea/miscellaneous.go]: indeed

**Comment by andrew-boyarshin** [comment]
I agree, but there were problems finding documentation. How can I be sure without docs that ```bind``` can bind to non-string fields? Will change.

**Comment by lunny** [comment]
@andrew-boyarshin what bind?

**Comment by andrew-boyarshin** [comment]
@lunny go-macaron one, the one which binds JSON GET request to ```api.MarkdownOption``` structure. ```go-macaron/binding.Bind```. Called in ```RegisterRoutes``` in ```routers/api/v1/api.go```.

**Comment by andrew-boyarshin** [comment]
@bkcsoft time to remove change request?

Pending merge.

**Comment by tboerger** [comment]
LGTM

**Pull request merged**